src/commands/publish.rs

```diff
@@ -116,17 +116,25 @@ pub struct PublishInitCommand {
 
 impl PublishInitCommand {
     /// Executes the command.
-    pub async fn exec(self) -> Result<()> {
+    pub async fn exec(self, retry: Option<Retry>) -> Result<()> {
         let config = self.common.read_config()?;
-        let client = self.common.create_client(&config)?;
+        let mut client = self.common.create_client(&config)?;
+        if let Some(retry) = retry {
+            retry.store_namespace(&client).await?
+        }
+        client.refresh_namespace(self.name.namespace()).await?;
 
+        let signing_key = if let Some(nm) = client.get_warg_header() {
+            self.common.signing_key(&RegistryUrl::new(nm.to_str()?)?)?
+        } else {
+            self.common.signing_key(client.url())?
+        };
```

Comment:
It looks like this snippets are repeated many times below. Can these two bits of logic be moved into `CommonOptions`? Would require some signature changes (or new methods) like:

```rust
pub async fn create_client(&self, config: &Config, retry: Option<Retry>) ...

pub fn signing_key(&self, client: &Client) -> Result<PrivateKey> {
```